controlplane/kubeadm/internal/etcd_client_generator.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internal
+
+import (
+	"crypto/tls"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/cluster-api/controlplane/kubeadm/internal/etcd"
+	"sigs.k8s.io/cluster-api/controlplane/kubeadm/internal/proxy"
+)
+
+// etcdClientGenerator generates etcd clients that connect to specific etcd members on particular control plane nodes.
+type etcdClientGenerator struct {
+	restConfig *rest.Config
+	tlsConfig  *tls.Config
+}
+
+func (c *etcdClientGenerator) clientFor(nodeName string) (*etcd.Client, error) {
```

Comment:
```suggestion
func (c *etcdClientGenerator) forNode(name string) (*etcd.Client, error) {
```

very minor: when I saw the usage above `etcdClientGenerator.clientFor` stutters

Reply:
ah yeah, good point